daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -36,42 +39,52 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
 
   def initial: State = State.empty
 
-  /** Invariants:
-    * - [[globalKeyInputs]]'s keyset is a superset of [[activeState]].[[ActiveLedgerState.keys]]'s keyset
-    *   and of all the [[ActiveLedgerState.keys]]'s keysets in [[rollbackStack]].
-    *   (the superset can be strict in case of by-key nodes inside an already completed Rollback scope)
-    * - In mode ![[ContractKeyUniquenessMode.Off]],
-    *   the keys belonging to the contracts in [[activeState]].[[ActiveLedgerState.consumedBy]]'s
-    *   keyset are in [[activeState]].[[ActiveLedgerState.keys]],
-    *   and similarly for all [[ActiveLedgerState]]s in [[rollbackStack]].
-    *
+  /**
     * @param locallyCreated
-    *   Tracks all contracts created by this transaction including those under a rollback.
+    *   Tracks all contracts created by an input node processed so far (including nodes under a rollback).
     *
     * @param globalKeyInputs
-    *   globalKeyInputs contains the key mapping required by Daml Engine to get to the current state.
-    *   It contains all keys that have been looked up during interpretation as well as entries for creates
-    *   (mapped to KeyCreate).
-    *   It can be used to resolve keys during re-interpretation.
+    *   Contains the key mapping required by Daml Engine to get to the current state.
```

Comment:
Based on the description at the start here, creates would not be included now. You only mention that at the very end. I think it makes sense to keep this mentioned in the high-level description as well.

Reply:
I've clarified that create nodes add a KeyCreate.